tensorboard/webapp/feature_flag/http/feature_flag_http_interceptor.ts

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {select, Store} from '@ngrx/store';
+import {Observable} from 'rxjs';
+import {first, switchMap} from 'rxjs/operators';
+import {getFeatureFlags} from '../store/feature_flag_selectors';
+import {State as FeatureFlagState} from '../store/feature_flag_types';
+
+export const FEATURE_FLAGS_HEADER_NAME = 'X-TensorBoard-Feature-Flags';
+
+/**
+ * HttpInterceptor for injecting feature flags into each HTTP request
+ * originating from the Angular TensorBoard code base.
+ */
+@Injectable({providedIn: 'root'})
```

Comment:
Why do we have providedIn: 'root' for this interceptor but the AuthHttpIntercepter just has @Injectable. TBH from my quick scan of the HttpInterceptor documentation this seems to make more sense but, I just don't understand the discrepancy.

Reply:
Good catch. At one point I was trying to follow the HttpInterceptor's "Usage Notes", including some of the other suggestions in there. Using `providedIn: 'root'` was one of the things. Ulimately I abandoned those guidelines so I may as well remove `providedIn: 'root'`, too.

Future me may regret this decision but it hasn't been a problem, as far as we know, for AuthHttpInterceptor in tbdev.